### PR TITLE
[app] Update terrain examples to use optimised RasterDemSource url.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes

Update terrain examples to use optimised RasterDemSource url.

`mapbox://mapbox.terrain-rgb` is the old source, pngs are larger and not optimised, this PR switches example to use `mapbox://mapbox.mapbox-terrain-dem-v1`. Note for this source, only `TileSize` of 514 is supported.

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt [hunk 1]
```diff
@@ -61,7 +61,7 @@ class SantaCatalinaActivity : AppCompatActivity() {
       style(styleUri = Style.SATELLITE_STREETS) {
         +rasterDemSource(SOURCE) {
           url(TERRAIN_URL_TILE_RESOURCE)
-          tileSize(512)
+          tileSize(514)
```
Comment (Chaoba) [inline app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt:64]: Just curious, why change to 514?

**Review by tobrun — APPROVED** [review]

File: app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt (hunk 1 above)
Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt:64]: More context at https://github.com/mapbox/mapbox-maps-ios/issues/470, I will also add a in-line comment here.

Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/examples/terrain/SantaCatalinaActivity.kt:64]:
```suggestion
          // 514 specifies padded DEM tile and provides better performance than 512 tiles.
          tileSize(514)
```

File: app/src/main/java/com/mapbox/maps/testapp/examples/terrain/Terrain3DShowcaseActivity.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ class Terrain3DShowcaseActivity : AppCompatActivity() {
       styleExtension = style(Style.SATELLITE_STREETS) {
         +rasterDemSource(SOURCE) {
           url(TERRAIN_URL_TILE_RESOURCE)
-          tileSize(512)
+          tileSize(514)
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/terrain/Terrain3DShowcaseActivity.kt:30]:
```suggestion
          // 514 specifies padded DEM tile and provides better performance than 512 tiles.
          tileSize(514)
```

**Review by Chaoba — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Pull request merged**